validator/lib/validateBoundingVolume.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+var Promise = require('bluebird');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var Cartesian3 = Cesium.Cartesian3;
+var Rectangle = Cesium.Rectangle;
+var Cartographic = Cesium.Cartographic;
+
+module.exports = validateVolume;
+
+function validateVolume(tileset) {
+    return new Promise(function(resolve) {
+        validateNode(tileset.root, tileset, resolve);
+    });
+}
+
+var scratchCartographic = new Cartographic();
+var scratchContentCartesian = new Cartesian3();
+var scratchTileCartesian = new Cartesian3();
+var scratchContentRectangle = new Rectangle();
+var scratchTileRectangle = new Rectangle();
+
+function regionInsideRegion(contentRectangle, tileRectangle) {
+    return (Rectangle.contains(tileRectangle,  Rectangle.northwest(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.southwest(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.northeast(contentRectangle, scratchCartographic))
+        && Rectangle.contains(tileRectangle, Rectangle.southeast(contentRectangle, scratchCartographic)));
+}
+
+function sphereInsideSphere(contentSphere, tileSphere) {
+    var contentRadius = contentSphere[3];
+    var tileRadius = tileSphere[3];
+    var contentCenter = Cartesian3.unpack(contentSphere, 0, scratchContentCartesian);
+    var tileCenter = Cartesian3.unpack(tileSphere, 0, scratchTileCartesian);
+    var distance = Cartesian3.distance(contentCenter, tileCenter);
+    return ((distance + contentRadius) <= tileRadius);
+}
+
+function validateNode(root, parent, resolve) {
+    var stack = [];
+    stack.push({
+        node: root,
+        parent: parent
+    });
+
+    while (stack.length > 0) {
+        var node = stack.pop();
+        var tile = node.node;
+        var parent = node.parent;
+        var tileContent = tile.content;
+
+        if (defined(tileContent) && defined(tileContent.boundingVolume)) {
+            if (defined(tileContent.boundingVolume.region) && defined(tile.boundingVolume.region)) {
+                var contentRegion = tileContent.boundingVolume.region;
+                var tileRegion = tile.boundingVolume.region;
+
+                var contentRect = Rectangle.unpack(contentRegion, 0, scratchContentRectangle);
+                var tileRect = Rectangle.unpack(tileRegion, 0, scratchTileRectangle);
+
+                var maxContentHeight = contentRegion[5];
+                var minContentHeight = contentRegion[4];
+                var maxTileHeight = tileRegion[5];
+                var minTileHeight = tileRegion[4];
+
+                if (!regionInsideRegion(contentRect, tileRect) || (maxContentHeight > maxTileHeight) || (minContentHeight < minTileHeight)) {
+                    return resolve({
```

Comment:
Similar to how `sphereInsideSphere` unpacks inside its function, you could also bring these lines into `regionInsideRegion` and pass in `contentRegion` and `tileRegion`. Then the rectangle check in `regionInsideRegion` can be factored out into a function called `rectangleInsideRectangle`.